### ISSUE TITLE
bdf cut script not working with realfast SDM

### ISSUE THREAD

**Issue by caseyjlaw** [demorest/sdmpy#18]
I am having using `bdf_dedisperse_cut.py` on an SDM from the realfast system.
The script hasn't been touched in a while, so I made a few small changes to using Python 3 syntax and skip xsd parsing. After that, I run:
`> bdf_dedisperse_cut.py -i 1:58:1 -d 959.3 -e ddcut realfast_19A-331.sb36775590.eb36779715.58647.98210152778_1560474781720`

and get:
```
  File "/Users/caseyjlaw/anaconda/envs/py3/lib/python3.7/site-packages/sdmpy-1.70.3-py3.7.egg/sdmpy/bdf.py", line 729, in __init__
    uid.replace(':/', '__').replace('/', '_'))
AttributeError: 'NoneType' object has no attribute 'replace'
```

I tried modifying the script to pass in a UID, but that leads to issues with the path to the bdf. I started to get the feeling that I wasn't understanding the setup right, so thought I'll file an issue.

**Comment by demorest** [comment]
Yeah, I think things must have changed in `BDFWriter` since we last ran this script.  I'll look into it today..

**Comment by demorest** [comment]
This filename error is fixed in 110b21d but there are still problems with this SDM.  There don't seem to be enough integrations saved to cover the full dispersion sweep.  I'll keep looking at it.

**Comment by demorest** [comment]
One more small bug, fixed in 37832d2.  This dataset works OK for me now.
